default_dialer.go

```diff
@@ -2,15 +2,17 @@ package enigma
 
 import (
 	"context"
-	"github.com/gorilla/websocket"
-	"github.com/pkg/errors"
 	"net"
 	"net/http"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
 )
 
 func setupDefaultDialer(dialer *Dialer) {
 	dialer.CreateSocket = func(ctx context.Context, url string, httpHeader http.Header) (Socket, error) {
 		gorillaDialer := websocket.Dialer{
+			Proxy:           http.ProxyFromEnvironment,
```

Comment:
Is it possible to override this?
Does it figure out itself if it's needed or not? e.g. If I want to use enigma to do internal calls and my org. has a Proxy defined for my computer

Reply:
Agree that this needs some kind of documentation on what it can be used for. Maybe just add an example in https://github.com/qlik-oss/enigma-go/tree/master/examples/authentication?

Reply:
Yes. It's possible.

Reply:
You can leverage environment variables to specify the proxy URL. https://pkg.go.dev/net/http#ProxyFromEnvironment. It picks up proxy URL directly from the environment variables. If you don't want to use proxy for a specific request. You can set NO_PROXY env variable.

Reply:
If you don't set any environment variables. `http.ProxyFromEnvironment` will just return nil.